bin/propolis-server/src/lib/vcpu_tasks.rs

```diff
@@ -187,7 +195,69 @@ impl VcpuTasks {
                         task.force_hold();
                         VmEntry::Run
                     }
-                    _ => {
+                    VmExitKind::InstEmul(inst) => {
+                        let diag = propolis::vcpu::Diagnostics::capture(vcpu);
+                        error!(log,
+                               "instruction emulation exit on vCPU {}",
+                               vcpu.id;
+                               "context" => ?inst,
+                               "vcpu_state" => %diag);
+
+                        event_handler.unhandled_vm_exit(vcpu.id, exit.kind);
+                        VmEntry::Run
+                    }
+                    VmExitKind::Unknown(code) => {
+                        error!(log,
+                               "unrecognized exit code on vCPU {}",
+                               vcpu.id;
+                               "code" => code);
+
+                        event_handler.unhandled_vm_exit(vcpu.id, exit.kind);
+                        VmEntry::Run
+                    }
+                    VmExitKind::Bogus => {
+                        error!(log,
+                               "lib returned bogus exit from vCPU {}",
+                               vcpu.id);
+
+                        event_handler.unhandled_vm_exit(vcpu.id, exit.kind);
+                        VmEntry::Run
+                    }
```

Comment:
Bogus is not an unhandled exit

Reply:
True, it's unconditionally handled in the lib. It previously fell into the `_` match arm, which called `unhandled_vm_exit`, but as long the lib is always supposed to handle this exit kind it's probably better just to panic directly here (since this is really more of an assertion failure than a truly unhandled exit).

Reply:
I would use `unreachable!`, then.  I think handling like this is misleading to the reader.

Reply:
Done in fba85be. I also tried to describe what the bogus exit is for in a comment. LMK if it's not accurate; I'm going mostly off what I could infer from reading around where VM_EXITCODE_BOGUS is used in bhyve.